Makefile

```diff
@@ -18,180 +18,68 @@
-test_preconditions:
-	@echo "Testing preconditions script"
-	@python test/scripts/preconditions/test_preconditions.py
-
-# Unit tests
-.PHONY: test_unit ## Run unit tests
-test_unit: test_migrate test_preconditions
-
-# Integration tests
-.PHONY: test_integration
-test_integration: ## Run integration tests
-	test/ci_integration.sh
-
-.PHONY: generate_docs
-generate_docs: ## Update README documentation for Terraform variables and outputs
-	@source test/make.sh && generate_docs
-
-.PHONY: release-new-version
-release-new-version:
-	@source helpers/release-new-version.sh
-
-# Run docker
+# Enter docker container for local development
 .PHONY: docker_run
-docker_run: ## Launch a shell within the Docker test environment
+docker_run:
 	docker run --rm -it \
-		-e BILLING_ACCOUNT_ID  \
-		-e DOMAIN \
-		-e FOLDER_ID \
-		-e GROUP_NAME \
-		-e ADMIN_ACCOUNT_EMAIL \
-		-e ORG_ID \
-		-e PROJECT_ID \
 		-e SERVICE_ACCOUNT_JSON \
-		-e GOOGLE_APPLICATION_CREDENTIALS=${CREDENTIALS_PATH} \
-		-v $(CURDIR):/cft/workdir \
-		${DOCKER_REPO_BASE_KITCHEN_TERRAFORM} \
-		/bin/bash -c "source test/ci_integration.sh && setup_environment && exec /bin/bash"
+		-v $(CURDIR):/workspace \
+		$(REGISTRY_URL)/${DOCKER_IMAGE_DEVELOPER_TOOLS}:${DOCKER_TAG_VERSION_DEVELOPER_TOOLS} \
+		/bin/bash
 
-.PHONY: docker_create
-docker_create: ## Run `kitchen create` within the Docker test environment
+# Execute prepare tests within the docker container
+.PHONY: docker_test_prepare
+docker_test_prepare:
 	docker run --rm -it \
-		-e BILLING_ACCOUNT_ID  \
-		-e DOMAIN \
-		-e FOLDER_ID \
-		-e GROUP_NAME \
-		-e ADMIN_ACCOUNT_EMAIL \
-		-e ORG_ID \
-		-e PROJECT_ID \
 		-e SERVICE_ACCOUNT_JSON \
-		-e GOOGLE_APPLICATION_CREDENTIALS=${CREDENTIALS_PATH} \
-		-v $(CURDIR):/cft/workdir \
-		${DOCKER_REPO_BASE_KITCHEN_TERRAFORM} \
-		/bin/bash -c "source test/ci_integration.sh && setup_environment && kitchen create"
-
-.PHONY: docker_converge
-docker_converge: ## Run `kitchen converge` within the Docker test environment
+		-e TF_VAR_org_id \
+		-e TF_VAR_folder_id \
+		-e TF_VAR_billing_account \
+		-v $(CURDIR):/workspace \
+		$(REGISTRY_URL)/${DOCKER_IMAGE_DEVELOPER_TOOLS}:${DOCKER_TAG_VERSION_DEVELOPER_TOOLS} \
+		/usr/local/bin/execute_with_credentials.sh prepare_environment
+
+# Clean up test environment within the docker container
+.PHONY: docker_test_cleanup
+docker_test_cleanup:
 	docker run --rm -it \
-		-e BILLING_ACCOUNT_ID  \
-		-e DOMAIN \
-		-e FOLDER_ID \
-		-e GROUP_NAME \
-		-e ADMIN_ACCOUNT_EMAIL \
-		-e ORG_ID \
-		-e PROJECT_ID \
 		-e SERVICE_ACCOUNT_JSON \
-		-e GOOGLE_APPLICATION_CREDENTIALS=${CREDENTIALS_PATH} \
-		-v $(CURDIR):/cft/workdir \
-		${DOCKER_REPO_BASE_KITCHEN_TERRAFORM} \
-		/bin/bash -c "source test/ci_integration.sh && setup_environment && kitchen converge"
-
-.PHONY: docker_verify
-docker_verify: ## Run `kitchen verify` within the Docker test environment
+		-e TF_VAR_org_id \
+		-e TF_VAR_folder_id \
+		-e TF_VAR_billing_account \
```

Comment:
This should be expanded to include:

- `TF_VAR_gsuite_admin_email`
- `TF_VAR_gsuite_domain`